From 24b74f9c65dc145dd6530afd9cd1059d04d8170f Mon Sep 17 00:00:00 2001
From: FintasticMan <finlay.neon.kid@gmail.com>
Date: Wed, 13 Nov 2024 00:29:08 +0100
Subject: [PATCH] Fix ioctl parameter type on musl

__musl__ doesn't exist, so the check doesn't correctly identify musl
systems.
---
 libavdevice/v4l2.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/libavdevice/v4l2.c b/libavdevice/v4l2.c
index 50ac47ec5a..0e6f8beae0 100644
--- a/libavdevice/v4l2.c
+++ b/libavdevice/v4l2.c
@@ -107,11 +107,7 @@ struct video_data {
     int (*open_f)(const char *file, int oflag, ...);
     int (*close_f)(int fd);
     int (*dup_f)(int fd);
-#if defined(__sun) || defined(__BIONIC__) || defined(__musl__) /* POSIX-like */
     int (*ioctl_f)(int fd, int request, ...);
-#else
-    int (*ioctl_f)(int fd, unsigned long int request, ...);
-#endif
     ssize_t (*read_f)(int fd, void *buffer, size_t n);
     void *(*mmap_f)(void *start, size_t length, int prot, int flags, int fd, int64_t offset);
     int (*munmap_f)(void *_start, size_t length);
-- 
2.47.0
